src/inlines/AutoLink.php

```diff
@@ -58,17 +58,22 @@ public static function consume(
 
     $uri = Str\slice($string, $start, $end - $start);
     if (\preg_match(self::ABSOLUTE_URI_PATTERN, $uri) === 1) {
-      return tuple(
-        new self($uri, $uri),
-        $offset,
-      );
+      if (!$context->areAllURISchemesEnabled()) {
+        $allowedURIs = $context->getAllowedURIs();
+        if (!C\any($allowedURIs, $elem ==> Str\starts_with_ci($uri, $elem))) {
```

Comment:
A previous comment mentioned that the `:` should not be considered part of the scheme - that's true, but does need to be handled here - e.g. `$elem.':'`